src/core/mac/sub_mac.hpp

```diff
@@ -602,10 +619,13 @@ class SubMac : public InstanceLocator
 #endif
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-    uint32_t  mCslTimeout;    ///< The CSL synchronized timeout in seconds.
-    TimeMicro mCslSampleTime; ///< The CSL sample time of the current period.
-    uint16_t  mCslPeriod;     ///< The CSL sample period, in units of 10 symbols (160 microseconds).
-    uint8_t   mCslChannel;    ///< The CSL sample channel.
+    uint32_t  mCslTimeout;     ///< The CSL synchronized timeout in seconds.
+    TimeMicro mCslSampleTime;  ///< The CSL sample time of the current period.
+    uint16_t  mCslPeriod;      ///< The CSL sample period, in units of 10 symbols (160 microseconds).
+    uint8_t   mCslChannel : 7; ///< The actually CSL sample channel. If `mCslChannelIsSet` is 0, this should be equal to
+                               ///< the Pan channel of `Mac`.
+    uint8_t
+        mIsCslChannelSpecified : 1; ///< Indicates that if CSL channel has been explicitly specified by the upper layer.
```

Comment:
Mac layer is also upper layer,  update to be more accurate?

```suggestion
        mIsCslChannelSpecified : 1; ///< Indicates whether or not the CSL channel was explicitly specified by the user.
```